src/Admin/AdherentAdmin.php

```diff
@@ -409,6 +370,83 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
                     $qb->leftJoin('managed_area.tags', 'managed_area_tag');
                     $qb->andWhere($qb->expr()->in('LOWER(managed_area_tag.name)', $value));
 
+                    return true;
+                },
+            ])
+            ->add('role', CallbackFilter::class, [
+                'label' => 'common.role',
+                'field_type' => ChoiceType::class,
+                'field_options' => [
+                    'choices' => AdherentRoleEnum::toArray(),
+                    'choice_label' => function (string $value) {
+                        return $value;
+                    },
+                    'multiple' => true,
+                ],
+                'show_filter' => true,
+                'callback' => function (ProxyQuery $qb, string $alias, string $field, array $value) {
+                    if (!$value['value']) {
+                        return false;
+                    }
+
+                    $where = new Expr\Orx();
+
+                    /** @var QueryBuilder $qb */
+
+                    // Referent
+                    if (\in_array(AdherentRoleEnum::REFERENT, $value['value'], true)) {
+                        $where->add(sprintf('%s.managedArea IS NOT NULL', $alias));
```

Comment:
To correspond to role definition in the `Adherent` class https://github.com/EnMarche/en-marche.fr/blob/master/src/Entity/Adherent.php#L774 we should check that tags exist.
Check other role definitions in this class.

Reply:
yes. in order to be a Referent,  `managedArea` should not be null (as stated here) AND `mangedArea.tags` should not be empty

Reply:
not need with this https://github.com/EnMarche/en-marche.fr/blame/8eee031dd79af478fabeeea4bdddeaf367976d89/src/Form/EventListener/ReferentManagedAreaListener.php#L29

managed_area_id become NULL if tags is empty